firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/Personalization.java

```diff
@@ -58,9 +74,29 @@ public void logArmActive(@NonNull String key, @NonNull ConfigContainer configCon
       return;
     }
 
+    String personalizationId = metadata.optString(PERSONALIZATION_ID);
+    if (personalizationId.isEmpty()) {
+      return;
+    }
+
+    synchronized (armsCache) {
+      if (armsCache.get(key) == personalizationId) {
+        return;
+      }
+      armsCache.put(key, personalizationId);
+    }
+
     Bundle params = new Bundle();
```

Comment:
Might call these `logParams` and `internalLogParams` since "parameters" otherwise refers to RC parameters.